rllib/evaluation/postprocessing.py

```diff
@@ -178,7 +178,24 @@ def compute_gae_for_sample_batch(
         input_dict = sample_batch.get_single_step_input_dict(
             policy.model.view_requirements, index="last"
         )
-        last_r = policy._value(**input_dict)
+
+        if policy.config["_enable_rl_module_api"]:
+            # TODO (Kourosh) This is a hack which I don't like at all. We need a
+            # permanent solution.
+            # Note: During sampling you are using the parameters at the beginning of
+            # the sampling process. If I'll be using this advantages during training
+            # should it not be the latest parameters during training for this to be
+            # correct? Does this mean that I need to preserve the trajectory
+            # information during training and compute the advantages inside the loss
+            # function?
```

Comment:
Got it. Yeah, we generally shuffle non-LSTM train batches prior to computing the loss (or splitting for SGD minibatching). True, this could potentially weaken learning success.

Reply:
IMPALA and v-trace keep the sequences in tact till the loss function. We could experiment doing this for PPO as well and then doing the postprocessing step on the learner side.

In general, I would like to move postprocessing off our samplers. Doesn't seem right there b/c it's not batched. But that would mean we would have to leave the trajectories in tact throughout our repay buffers (which could work, but it'll require some work).

Reply:
yeah, I don't know if it matters in practice. It doesn't seem like that it matters.